### ISSUE TITLE
logUrl in 'load' function in load.py no longer works

### ISSUE THREAD

**Issue by perryraskin** [yyss8/cunyfirst-login-with-selenium#1]
How did you figure out what the link is? I'd like to try to help fix this!

**Comment by yyss8** [comment]
Hi, perryraskin

it has been a while since I developed this project, it seems like they've updated their login system.

the new login url should be now: 

https://ssologin.cuny.edu/oam/server/auth_cred_submit?username=fakeUsername&password=fakePassword&submit.x=0&submit.y=0 

you can replace the username and password with your own account info or %s if you want to have them as input in the python console
However, when I tried to access it from a browser, it denied me with error code OAM-5:

OAM-5 | The user account is locked or disabled. Please contact the System Administrator.

I'm still able to log in by entering username and password on their login page tho, which means there might some hidden HTTP header that you need to add into the request to be able to login

**Comment by perryraskin** [comment]
I'm just getting a "failed to load" when i try to load that link. Is there a better way to login to the website? A different way than using the URL?

**Comment by yyss8** [comment]
Did you replace the new link yet?  I think you might have to add the some http headers to be able to fake the request 

here's what I found when my browser sends a request to their server

Connection:Keep-Alive
Content-Length:1374
Content-Type:text/html; charset=iso-8859-1
Date:Sun, 27 Aug 2017 18:59:45 GMT
Keep-Alive:timeout=5, max=100
Location:https://home.cunyfirst.cuny.edu/obrar.cgi?encreply=35Dg7wqcEdXs8qpu1zAcWXo+D3mlE8qB0Z8yQ/f7eEcKAjIwk8oc3VUWRB2gllUjKsL/CJA+he7VzxL9ZiiKjnpq7NDL7EOCZgyk6gMzcWv49XwH3V31w9L5YeRhxKySohlV1ObmhgaW2tddmOr6FUBLsMFoUV6R2PbF9pt9jxLP7TDQabjw1VttKLFClH1tS5DR7e2aNbBapBpkeOXNaGRjoWHtVs5cjPmPRvd5ooKM3fbYJX9f9pkAUWQCZUtzRswtqpRaMd0mSccYsNkF0mjPN7S1r6ln/j0jTmgCl/iUqjOtQMR59S4hYbUTXafOOWRO3F0BysYGuu5Q7oY5dF4HjJXjUSPv2t87mSicEp0QiKtkO9bL7Gvxe0rml9D+DRHhwlFj5Y4D0KpuFN8Bs67EuJtjsJbkTtHZFLZbP4UCo8UFQrgVvLvOxK3F2rWXzfJruQvSX0kgM01i5CROBjqCvBF/UEAuN1/iTGUaSeYLuNlOqIlTTVSxz6tiBO6w+3k/rgURZFKY5/arqKaFPyldVCfBRNyDB6H51jhxLv8EA8LPelpEexZslGs14vDjSFckIVUUX1aqI/bzv2OuYWVvrU8Fwr4Wfz+rWQwMvdpF5sacZlbI2SBmY43bPxZFKK6OG7GKejyc0BLEzd59tcl1CMmuYERBX1Ypns97XRae+6/kEs7X9eTqxJr4Ju6bTgiSJ//rkwKPAOt+00vw0PGk8XvETTzBeQQYUCBpiqAm9TwjU1OnyGsFPfWq0CmpN+Q1MISJtZiftZzUnsn+bYIKheKGKWwZrGm84iGwq9dHiSzPPRRYB26BD7YgipMz97PJT2SNv4jZIbpvRRJz66zKS58AMfR8wnICBZQOYxNowpH7R2oB/F4qFVLI+/f+b0PPWq/lz1hxTOIWoD//zUMy9cDhTALvyL6AxdapzQ0z/XSNYOBDQ2HamtWehA4G36WDdcCfhxrVg7nsddArl7Fm1Bieol3hXYb3T8rpc5/pqXEbEPz2w2IuOumSk8r8xNjQhC6u1jbH7hPtetFAWj1R1x+VV+qHPYmYMLoP78TA9llWj1b9/a8QoptD/T+RUiHXfi9YFBkRWFK+VyuhBw==
Server:Oracle-HTTP-Server-11g

location seems to be key header that has been to included in the request, otherwise it would returns error code 503

**Comment by perryraskin** [comment]
I did not replace it in the code, I tried to load that link in my browser though

**Comment by perryraskin** [comment]
When I replace the code in load.py and run it, I get "urllib.error.HTTPError: HTTP Error 503: Service Temporarily Unavailable" after I try to login through the program

**Comment by yyss8** [comment]
Yeah, I tried it as well, I could use the previous link to log into the system directly

Error 503 is what I expected  if you try to log in from Python, because the request doesn't include the headers the server requests.

Well, I think you can still use the selenium library to log into their system
take a look on Search.py file and replace the link in that file with the login page url and perhaps some HTML element as well like the button class to see if it still works.

**Comment by perryraskin** [comment]
I'm getting other errors, like 'phantomjs' executable needs to be in PATH

**Comment by yyss8** [comment]
try to comment out  webdriver.PhantomJS() and uncomment webdriver.Chrome()

        browser = webdriver.PhantomJS()
        #browser = webdriver.Firefox()
        #browser = webdriver.Chrome()

**Comment by perryraskin** [comment]
Now I'm getting: selenium.common.exceptions.WebDriverException: Message: 'chromedriver' executable needs to be in PATH. Please see https://sites.google.com/a/chromium.org/chromedriver/home

**Comment by perryraskin** [comment]
i downloaded the chrome driver file, but now it opens Chrome and doesnt do anything else, the website doesnt even load. i don't want it to open chrome, though

**Comment by yyss8** [comment]
Well, since Selenium is only a browser simulator, it's able to fake the browser so you wouldn't need to deal anything with the annoying headers.
that's why I used PhantomJS which is also a browser but only running in background
you can download it from their website and put it in the same folder as the pythong script

**Comment by perryraskin** [comment]
Ok, I put the PhantomJS file in the bin folder, but now I keep getting more errors:
`selenium.common.exceptions.NoSuchElementException: Message: {"errorMessage":"Unable to find element with id 'crefli_HC_SSS_STUDENT_CENTER'","request":{"headers":{"Accept":"application/json","Accept-Encoding":"identity","Connection":"close","Content-Length":"109","Content-Type":"application/json;charset=UTF-8","Host":"127.0.0.1:63581","User-Agent":"Python http auth"},"httpVersion":"1.1","method":"POST","post":"{\"using\": \"id\", \"value\": \"crefli_HC_SSS_STUDENT_CENTER\", \"sessionId\": \"34778330-8c1c-11e7-89e6-b77fda07e361\"}","url":"/element","urlParsed":{"anchor":"","query":"","file":"element","directory":"/","path":"/element","relative":"/element","port":"","host":"","password":"","user":"","userInfo":"","authority":"","protocol":"","source":"/element","queryKey":{},"chunks":["element"]},"urlOriginal":"/session/34778330-8c1c-11e7-89e6-b77fda07e361/element"}}
Screenshot: available via screen`

**Comment by perryraskin** [comment]
I think the issue is that the "Login" button never gets clicked. The name is still 'submit' so I am not sure how to fix it

**Comment by yyss8** [comment]
I've reinstalled python and tested my codes
It seems like PhantomJS is having issue finding the elements but works fine with Firefox and Chrome

here's the code that I used to test the script

search = ClsSearch()
search.id = "FakeUserName"
search.pw = "FakePassword"
loginStatus = search.login()

lastMenu = False
while lastMenu != True:  
    print("")
    print("1- Search One Class")
    print("2- Search All Class")
    print("3- Class Analysis")
    print("4- Back")
    if loginStatus == True:
        ftSelected = input("Please select the features: ")
        if ftSelected != "3" and ftSelected != "4": #Initialize search feature for first two options
            collegeList = search.loadCollegeList()
            i = 1
            print("")
            clgSorted = sorted(collegeList)
            for college in clgSorted:
                print(str(i) + " -  "+ collegeList[college]) 
                i += 1
            print("")
            clgSelected = input("Please select the college: ")
            search.selectCollege(clgSorted[int(clgSelected) - 1])
            print("")        
            print("1 - Spring 2016")
            print("2 - Summer 2016")
            print("3 - Fall 2016")
            print("")
            smtSelected = input("Please select the semster: ")
            if smtSelected == "1":
                term = "1162"
            elif smtSelected == "2":
                term = "1166"
            elif smtSelected == "3":
                term = "1169"
            else:
                term = ""
                print("In-correct Input!")
        
            if term != "":
                search.loadTerm(term)
                if ftSelected == "1":
                    search.search_One_Class()
                elif ftSelected == "2":
                    clsSearchAll(search)
        elif ftSelected == "3":clsAnalyze()
        elif ftSelected == "4":
            lastMenu = True
            search.logOut()
        else:print("In-correct Input!")
    else:
        print("Login Failed")
        lastMenu = True
        search.logOut()

**Comment by perryraskin** [comment]
When I try Chrome/Firefox, nothing happens, and I get errors. Chrome opens as a blank page.

`selenium.common.exceptions.WebDriverException: Message: unknown error: Runtime.executionContextCreated has invalid 'context': {"auxData":{"frameId":"18630.1","isDefault":true},"id":1,"name":"","origin":"://"}
  (Session info: chrome=61.0.3163.59)
  (Driver info: chromedriver=2.9.248307,platform=Mac OS X 10.12.6 x86_64)`
